app/wasm/query_plugin.go

```diff
@@ -0,0 +1,105 @@
+package wasm
+
+import (
+	"encoding/json"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	wasmvmtypes "github.com/CosmWasm/wasmvm/types"
+
+	bindings "github.com/osmosis-labs/osmosis/v7/app/wasm/bindings"
+)
+
+func CustomQuerier(osmoKeeper *QueryPlugin) func(ctx sdk.Context, request json.RawMessage) ([]byte, error) {
```

Comment:
This is the entry point to handle queries. 
If you add another variant to `OsmosisQuery`, you can add the dispatch code here. (and the actual logic to `QueryPlugin`). 
We need this structure as the API takes a function for querier, not an interface.